ably/rest_channel.go

```diff
@@ -52,23 +52,29 @@ func newRESTChannel(name string, client *REST) *RESTChannel {
 }
 
 // Publish publishes a message on the channel.
-func (c *RESTChannel) Publish(ctx context.Context, name string, data interface{}) error {
-	return c.PublishMultiple(ctx, []*Message{
-		{Name: name, Data: data},
-	})
+func (c *RESTChannel) Publish(ctx context.Context, name string, data interface{}, options ...PublishMultipleOption) error {
+	return c.PublishMultiple(ctx, []*Message{{Name: name, Data: data}}, options...)
 }
 
 // PublishMultiple publishes multiple messages in a batch.
-func (c *RESTChannel) PublishMultiple(ctx context.Context, messages []*Message) error {
-	return c.PublishMultipleWithOptions(ctx, messages)
+func (c *RESTChannel) PublishMultiple(ctx context.Context, messages []*Message, options ...PublishMultipleOption) error {
```

Comment:
There doesn't seem to be any difference in the method signature of ` PublishMultiple` and `PublishMultipleWithOptions`. 
I don't think we need to change code at `60-61`. Instead add one more method `PublishWithOptions`. So, we will have 4 methods in total.
```
1. Publish -> Calls PublishMultiple
2. PublishWithOptions -> Calls PublishMultipleWithOptions
3. PublishMultiple -> Calls PublishMultipleWithOptions with no options 
4. PublishMultipleWithOptions -> Has real business logic as is, no need to change it
```

Reply:
If we don't want to have `WithOptions` methods, we can just keep **`first`** and **`third`**, accepting optional params at the end.

Reply:
If we do decide to drop the `PublishMultipleWithOptions` function (which I agree will now be redundant), we need to deprecate it and advise what the alternative is, rather than removing it straight away.

Reply:
Yeah, that's what I meant. Since it's a public interface, we will deprecate it for the time being. Thanks for the reminder @lmars.

Reply:
If you are happy @sacOO7 please can you resolve this conversation.

Reply:
No, I think we need to make changes. Can you go through the conversation again?

Reply:
What are the changes you think we need to make here?

Reply:
I suggest having four methods as mentioned. It's more explicit and verbose for SDK users.

Reply:
As this is a bug fix for an issue a customer is facing, I would advocate for making the smallest amount of change possible to resolve the issue.

Reply:
Resolving this conversation on @sacOO7's behalf so that @Rosalita can land this PR.

Reply:
I did the deprecation in https://github.com/ably/ably-go/pull/450.